components/shared/onboarding/Onboarding.tsx

```diff
@@ -0,0 +1,320 @@
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import OnboardingThree from "./OnboardingThree";
+import { updateUser } from "@/lib/actions/user.actions";
+import OnboardingFour from "./OnboardingFour";
+import CustomButton from "../CustomButton";
+import { Progress } from "@/components/ui/progress";
+import OnboardingTick from "./OnboardingTick";
+import UploadPhoto from "../profile/edit/UploadPhoto";
+import { useRouter } from "next/navigation";
+
+interface OnboardingProps {
+  children?: React.ReactNode;
+  step: string;
+  user: Partial<IUser>;
+}
+
+const Onboarding = ({ step, user }: OnboardingProps) => {
+  const [currentStep, setCurrentStep] = React.useState(step);
+  const router = useRouter();
+  const [pending, startTransition] = useTransition();
+  const {
+    handleSubmit,
+    control,
+    register,
+    getValues,
+    formState: { errors },
+  } = useForm<z.infer<typeof CompleteProfileEditSchema>>({
+    resolver: zodResolver(CompleteProfileEditSchema),
+    defaultValues: {
+      learningGoals: user.learningGoals || [],
+      experiences: user.experiences?.map((exp) => ({ name: exp })) || [],
+      availability: {
+        startTime: user?.startTime || new Date(),
+        endTime: user?.endTime || new Date(),
+      },
+      newProjects: user?.newProjects || false,
+      technologies: user?.technologies || [],
+      portfolio: user?.portfolio || "",
+      email: user?.email || "",
+    },
+  });
+  const {
+    fields: goalFields,
+    append: appendGoal,
+    remove: removeGoal,
+  } = useFieldArray<z.infer<typeof CompleteProfileEditSchema>>({
+    control,
+    name: "learningGoals",
+  });
+
+  const {
+    fields: experienceFields,
+    append: appendExperience,
+    remove: removeExperience,
+  } = useFieldArray<z.infer<typeof CompleteProfileEditSchema>>({
+    control,
+    name: "experiences",
+  });
+  const changeStep = async () => {
+    const nextStep = parseInt(currentStep) + 1;
+    const {
+      fullname,
+      portfolio,
+      availability,
+      learningGoals,
+      email,
+      experiences,
+      newProjects,
+    } = getValues();
+    try {
+      if (nextStep === 2) {
+        await updateUser({
+          updateData: {
+            fullname,
+            portfolio,
+            email,
+          },
+        });
+      }
+      if (nextStep === 3) {
+        await updateUser({
+          updateData: {
+            learningGoals,
+          },
+        });
+      }
+      if (nextStep === 4) {
+        await updateUser({
+          updateData: {
+            experiences: experiences?.map((experience) => experience.name),
+          },
+        });
+      }
+      if (nextStep === 5) {
+        startTransition(() => {
+          updateUser({
+            updateData: {
+              newProjects,
+              startTime: availability?.startTime,
+              endTime: availability?.endTime,
+            },
+          });
+        });
+      }
```

Comment:
This 'changeStep' should not be running on step 5, right? Because we don't render the button that does this? Just the submit. So this step doesn't belong here anyway.

If we want to use transition with an awaited value- we should make it async and await it. Otherwise it completes immediately and doesn't give us a "usePending" state.